patches/net/minecraft/client/gui/Gui.java.patch

```diff
@@ -237,6 +237,15 @@
              }
          }
  
+@@ -664,7 +_,7 @@
+     }
+ 
+     private void renderDemoOverlay(GuiGraphics p_281825_, float p_316211_) {
+-        if (this.minecraft.isDemo()) {
++        if (this.minecraft.isDemo() && !this.getDebugOverlay().showDebugScreen()) { // NEO: Hide demo timer when F3 debug overlay is open
```

Comment:
Please add a reference to the Mojira issue: https://bugs.mojang.com/browse/MC-271166